### PR TITLE
Fix the bug in the MacroscopicEvolveM.cpp 

### PR DESCRIPTION
**Pull request by ruohai0925**
While updating the M field,  M appears in both the RHS and LHS, which is not consistent with the formula.  

A temporary MultiFab Mfield_prev is built , copied from the MultiFab Mfield, and used in the RHS of the updated equation. Similar to the method used in the MacroscopicEvolveM_2nd.cpp 

Here is a test for the coupled equations. 

`################################
#################################
max_step = 10
amr.n_cell = 8 8 8 # number of cells spanning the domain in each coordinate direction at level 0
amr.max_grid_size = 512 # maximum size of each AMReX box, used to decompose the domain
amr.blocking_factor = 8
geometry.coord_sys   = 0
geometry.is_periodic = 1  1  1
geometry.prob_lo     =  -1.5e-6 -1.5e-6 -1.5e-6
geometry.prob_hi     =   1.5e-6  1.5e-6  1.5e-6
amr.max_level = 0
#################################
#################################
warpx.verbose = 0
warpx.use_filter = 0
warpx.cfl = 4000
warpx.do_pml = 0

warpx.mag_time_scheme_order = 2 # default 1
macroscopic.mag_normalized_error = 0.1 # default 0.1
macroscopic.mag_max_iter = 100 # default 100
macroscopic.mag_tol = 0.0001 # default 0.0001
warpx.safe_guard_cells = 0 # default 0

#algo.em_solver_medium = vacuum # vacuum/macroscopic
algo.em_solver_medium = macroscopic # vacuum/macroscopic
algo.macroscopic_sigma_method = laxwendroff # laxwendroff or backwardeuler
macroscopic.sigma_init_style = "parse_sigma_function" # parse or "constant"
macroscopic.sigma_function(x,y,z) = "0.0"
#macroscopic.sigma = 0.0
macroscopic.epsilon_init_style = "parse_epsilon_function" # parse or "constant"
macroscopic.epsilon_function(x,y,z) = "8.8541878128e-12"
#macroscopic.epsilon = 8.8541878128e-12
macroscopic.mu_init_style = "parse_mu_function" # parse or "constant"
macroscopic.mu_function(x,y,z) = "1.25663706212e-06"
#macroscopic.mu = 1.25663706212e-06
macroscopic.mag_Ms_init_style = "parse_mag_Ms_function" # parse or "constant"
macroscopic.mag_Ms_function(x,y,z) = "1.4e5"
#macroscopic.mag_Ms = 1.4e5 # in unit A/m, equal to 1750 Gauss
#unit conversion: 1 Gauss = (1000/4pi) A/m
macroscopic.mag_alpha_init_style = "parse_mag_alpha_function" 
macroscopic.mag_alpha_function(x,y,z) = "5e-01"
#macroscopic.mag_alpha = 5e-02 # alpha is unitless, typical values range from 1e-3 ~ 1e-5
macroscopic.mag_gamma_init_style = "parse_mag_gamma_function"
macroscopic.mag_gamma_function(x,y,z) = "-1.759e11"
#macroscopic.mag_gamma = -1.759e11 # gyromagnetic ratio is NEGATIVE constant for electrons in all materials
particles.nspecies = 0
#################################
#################################
my_constants.pi = 3.14159265359
my_constants.L = 141.4213562373095e-6
my_constants.c = 299792458.
my_constants.wavelength = 1.2e-1
warpx.E_ext_grid_init_style = parse_E_ext_grid_function
warpx.Ez_external_grid_function(x,y,z) = 0.
warpx.Ex_external_grid_function(x,y,z) = 1e4 
warpx.Ey_external_grid_function(x,y,z) = 0.
#warpx.Ey_external_grid_function(x,y,z) = "1.e5*exp(-z**2/L**2)*cos(2*pi*z/wavelength)"
warpx.B_ext_grid_init_style = parse_B_ext_grid_function
#warpx.Bx_external_grid_function(x,y,z)= "-1.e5*exp(-z**2/L**2)*cos(2*pi*z/wavelength)/c"
warpx.Bx_external_grid_function(x,y,z)= 0.0
#warpx.Bx_external_grid_function(x,y,z)= 0.
warpx.By_external_grid_function(x,y,z)= 0.
warpx.Bz_external_grid_function(x,y,z) = 0.
warpx.H_bias_ext_grid_init_style = parse_H_bias_ext_grid_function
warpx.Hx_bias_external_grid_function(x,y,z)= 0.
warpx.Hy_bias_external_grid_function(x,y,z)= 0.
warpx.Hz_bias_external_grid_function(x,y,z)= 3e4 # in A/m, equal to 382 Oersted
#warpx.Hz_bias_external_grid_function(x,y,z)= 0. # in A/m, equal to 382 Oersted
#unit conversion: 1 Gauss = 1 Oersted = (1000/4pi) A/m
#calculation of H_bias: H_bias (oe) = frequency / 2.8e6
#warpx.H_bias_ext_grid_init_style = constant
#warpx.Hx_bias_external_grid = 0
#warpx.Hy_bias_external_grid = 0
#warpx.Hz_bias_external_grid = 0
#warpx.M_ext_grid_init_style = parse_M_ext_grid_function
#warpx.Mx_external_grid_function(x,y,z)= "-1.e5*exp(-z**2/L**2)*cos(2*pi*z/wavelength)/c"
#warpx.My_external_grid_function(x,y,z)= 0.
#warpx.Mz_external_grid_function(x,y,z) = 0.
warpx.M_ext_grid_init_style = constant
warpx.M_external_grid = 140000. 0. 0.
#Diagnostics
diagnostics.diags_names = plt
plt.period = 1
plt.diag_type = Full
plt.fields_to_plot = Mx_xface My_xface Mz_xface Mx_yface My_yface Mz_yface Mx_zface My_zface Mz_zface
#plt.fields_to_plot = Mx_xface My_xface Mz_xface Ex Ey Ez Bx By Bz
plt.plot_raw_fields = 0`

### CODE REVIEW

File: Source/FieldSolver/FiniteDifferenceSolver/EvolveB.cpp [hunk 1]
```diff
@@ -107,7 +107,7 @@ void FiniteDifferenceSolver::EvolveBCartesian (
                              - dt * T_Algo::UpwardDx(Ey, coefs_x, n_coefs_x, i, j, k);
             }
 
-        );
+	);
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/EvolveB.cpp]:
```suggestion
        );
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp [hunk 2]
```diff
@@ -114,30 +125,30 @@ void FiniteDifferenceSolver::MacroscopicEvolveM (
...
+                - M_xface_prev(i, j, k, 1) * ( M_xface_prev(i, j, k, 1) * Hz_eff - M_xface_prev(i, j, k, 2) * Hy_eff));
 
 
               // temporary normalized magnitude of M_xface field at the fixed point
               // re-investigate the way we do Ms interp, in case we encounter the case where Ms changes across two adjacent cells that you are doing interp
               amrex::Real mag_normalized = std::sqrt( std::pow(M_xface(i, j, k, 0),2.0) + std::pow(M_xface(i, j, k, 1),2.0) +
                       std::pow(M_xface(i, j, k, 2),2.0) ) / MacroscopicProperties::macro_avg_to_face(i,j,k,amrex::IntVect(1,0,0),mag_Ms_arr);
 
-              // check the normalized error
+	      // check the normalized error
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp]:
```suggestion
	        // check the normalized error
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp [hunk 3]
```diff
@@ -91,19 +101,20 @@ void FiniteDifferenceSolver::MacroscopicEvolveM (
...
               Real Hz_bias_xface = MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(0,0,1), amrex::IntVect(1,0,0), Hz_bias);
               // H_eff = H_maxwell + H_bias + H_exchange + H_anisotropy ... (only the first two terms are considered here)
-              Real Hx_eff = Hx_bias_xface;
-              Real Hy_eff = Hy_bias_xface;
-              Real Hz_eff = Hz_bias_xface;
+              // Real Hx_eff = Hx_bias_xface;
+              // Real Hy_eff = Hy_bias_xface;
+              // Real Hz_eff = Hz_bias_xface;
 
               // keep in mind to ADD H_Maxwell back to H_eff as shown below
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp]:
```suggestion
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp [hunk 4]
```diff
@@ -91,19 +101,20 @@ void FiniteDifferenceSolver::MacroscopicEvolveM (
               Real Hx_xface = MacroscopicProperties::getH_Maxwell(i, j, k, 0, amrex::IntVect(1,0,0), amrex::IntVect(1,0,0), Bx, M_xface);
               Real Hy_xface = MacroscopicProperties::getH_Maxwell(i, j, k, 1, amrex::IntVect(0,1,0), amrex::IntVect(1,0,0), By, M_xface);
               Real Hz_xface = MacroscopicProperties::getH_Maxwell(i, j, k, 2, amrex::IntVect(0,0,1), amrex::IntVect(1,0,0), Bz, M_xface);
-              // H_bias
+
+	      // H_bias
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp]:
```suggestion
	        // H_bias
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp [hunk 5]
```diff
@@ -44,7 +44,14 @@ void FiniteDifferenceSolver::MacroscopicEvolveM (
         std::unique_ptr<MacroscopicProperties> const& macroscopic_properties )
     {
 
-        // obtain the maximum relative amount we let M deviate from Ms before aborting
+        // build temporary vector<multifab,3> Mfield_prev
+        std::array< std::unique_ptr<amrex::MultiFab>, 3 > Mfield_prev; // Mfield data in previous step
+        for (int i = 0; i < 3; i++){
+        Mfield_prev[i].reset( new MultiFab(Mfield[i]->boxArray(),Mfield[i]->DistributionMap(),3,Mfield[i]->nGrow()));
+        MultiFab::Copy(*Mfield_prev[i],*Mfield[i],0,0,3,Mfield[i]->nGrow());
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp]:
```suggestion
            Mfield_prev[i].reset( new MultiFab(Mfield[i]->boxArray(),Mfield[i]->DistributionMap(),3,Mfield[i]->nGrow()));
            MultiFab::Copy(*Mfield_prev[i],*Mfield[i],0,0,3,Mfield[i]->nGrow());
```

File: Source/Evolve/WarpXEvolve.cpp [hunk 6]
```diff
@@ -421,6 +421,12 @@ WarpX::OneStep_nosub (Real cur_time)
 #endif //
 #endif // end for PSATD
     }
+
+#ifdef WARPX_MAG_LLG
+    // output the field variables on level 0
+    MacroscopicfieldOutput(Mfield_fp[0], Efield_fp[0], Bfield_fp[0], cur_time);
+#endif
+
 }
```
Comment (ajnonaka) [inline Source/Evolve/WarpXEvolve.cpp]: Is this for debugging only?  If so we should remove this.

File: Source/WarpX.cpp [hunk 7]
```diff
@@ -1643,3 +1643,82 @@ WarpX::PicsarVersion ()
...
+              },
+
+            [=] AMREX_GPU_DEVICE (int i, int j, int k){
+
+              });
+        }
+    }
+#endif
+
+
```
Comment (ajnonaka) [inline Source/WarpX.cpp]: We should discuss with Reva how to handle these types of debugging functions (in terms of adding them to our primary branch)

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp [hunk 8]
```diff
@@ -490,23 +490,23 @@ void FiniteDifferenceSolver::MacroscopicEvolveM_2nd (
...
               Real Hz_bias_zface = MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(0,0,1), amrex::IntVect(0,0,1), Hz_bias);
               // H_eff = H_maxwell + H_bias + H_exchange + H_anisotropy ... (only the first two terms are considered here)
-              Real Hx_eff = Hx_bias_zface;
-              Real Hy_eff = Hy_bias_zface;
-              Real Hz_eff = Hz_bias_zface;
+              // Real Hx_eff = Hx_bias_zface;
+              // Real Hy_eff = Hy_bias_zface;
+              // Real Hz_eff = Hz_bias_zface;
 
               // Uncomment these lines when we couple LLG with Maxwell
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp]:
```suggestion
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp [hunk 9]
```diff
@@ -231,23 +231,23 @@ void FiniteDifferenceSolver::MacroscopicEvolveM_2nd (
...
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp:247]:
```suggestion
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp [hunk 10]
```diff
@@ -361,23 +361,23 @@ void FiniteDifferenceSolver::MacroscopicEvolveM_2nd (
...
               Real Hz_bias_xface = MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(0,0,1), amrex::IntVect(1,0,0), Hz_bias);
               // H_eff = H_maxwell + H_bias + H_exchange + H_anisotropy ... (only the first two terms are considered here)
-              Real Hx_eff = Hx_bias_xface;
-              Real Hy_eff = Hy_bias_xface;
-              Real Hz_eff = Hz_bias_xface;
+              // Real Hx_eff = Hx_bias_xface;
+              // Real Hy_eff = Hy_bias_xface;
+              // Real Hz_eff = Hz_bias_xface;
 
               // Uncomment these lines when we couple LLG with Maxwell
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp]:
```suggestion
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp [hunk 11]
```diff
@@ -361,23 +361,23 @@ void FiniteDifferenceSolver::MacroscopicEvolveM_2nd (
...
               // // H_maxwell
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp]:
```suggestion
              // H_maxwell
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp [hunk 12]
```diff
@@ -231,23 +231,23 @@ void FiniteDifferenceSolver::MacroscopicEvolveM_2nd (
...
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp]:
```suggestion
              // H_maxwell
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp (hunk 9 above)
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp]:
```suggestion
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp [hunk 13]
```diff
@@ -174,24 +174,24 @@ void FiniteDifferenceSolver::MacroscopicEvolveM_2nd (
...
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp]:
```suggestion
              // H_maxwell
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp [hunk 14]
```diff
@@ -118,23 +118,23 @@ void FiniteDifferenceSolver::MacroscopicEvolveM_2nd (
...
               Real Hz_bias_xface = MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(0,0,1), amrex::IntVect(1,0,0), Hz_bias);
               // H_eff = H_maxwell + H_bias + H_exchange + H_anisotropy ... (only the first two terms are considered here)
-              Real Hx_eff = Hx_bias_xface;
-              Real Hy_eff = Hy_bias_xface;
-              Real Hz_eff = Hz_bias_xface;
+              // Real Hx_eff = Hx_bias_xface;
+              // Real Hy_eff = Hy_bias_xface;
+              // Real Hz_eff = Hz_bias_xface;
 
               // Uncomment these lines when we couple LLG with Maxwell
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp]:
```suggestion
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp [hunk 15]
```diff
@@ -118,23 +118,23 @@ void FiniteDifferenceSolver::MacroscopicEvolveM_2nd (
...
               // // H_maxwell
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp]:
```suggestion
              // H_maxwell
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp [hunk 16]
```diff
@@ -212,21 +223,22 @@ void FiniteDifferenceSolver::MacroscopicEvolveM (
...
               Real Hy_zface = MacroscopicProperties::getH_Maxwell(i, j, k, 1, amrex::IntVect(0,1,0), amrex::IntVect(0,0,1), By, M_zface);
               Real Hz_zface = MacroscopicProperties::getH_Maxwell(i, j, k, 2, amrex::IntVect(0,0,1), amrex::IntVect(0,0,1), Bz, M_zface);
-              // H_bias
+
+	      // H_bias
               Real Hx_bias_zface = MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(1,0,0), amrex::IntVect(0,0,1), Hx_bias);
               Real Hy_bias_zface = MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(0,1,0), amrex::IntVect(0,0,1), Hy_bias);
               Real Hz_bias_zface = MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(0,0,1), amrex::IntVect(0,0,1), Hz_bias);
               // H_eff = H_maxwell + H_bias + H_exchange + H_anisotropy ... (only the first two terms are considered here)
               // keep in mind to ADD H_Maxwell back to H_eff as shown below
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp]:
```suggestion
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp [hunk 17]
```diff
@@ -212,21 +223,22 @@ void FiniteDifferenceSolver::MacroscopicEvolveM (
...
-              // Real Hz_eff = Hz_zface + Hz_bias_zface;
-              Real Hx_eff = Hx_bias_zface;
-              Real Hy_eff = Hy_bias_zface;
-              Real Hz_eff = Hz_bias_zface;
+              Real Hx_eff = Hx_zface + Hx_bias_zface;
+              Real Hy_eff = Hy_zface + Hy_bias_zface;
+              Real Hz_eff = Hz_zface + Hz_bias_zface;
+              // Real Hx_eff = Hx_bias_zface;
+              // Real Hy_eff = Hy_bias_zface;
+              // Real Hz_eff = Hz_bias_zface;
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp]:
```suggestion
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp [hunk 18]
```diff
@@ -212,21 +223,22 @@ void FiniteDifferenceSolver::MacroscopicEvolveM (
               // when working on M_zface(i,j,k,0:2) we have direct access to M_zface(i,j,k,0:2) and Hz(i,j,k)
               // Hy and Hz can be acquired by interpolation
               // H_maxwell
-              Real Hx_zface = MacroscopicProperties::getH_Maxwell(i, j, k, 0, amrex::IntVect(1,0,0), amrex::IntVect(0,0,1), Bx, M_zface);
+	      Real Hx_zface = MacroscopicProperties::getH_Maxwell(i, j, k, 0, amrex::IntVect(1,0,0), amrex::IntVect(0,0,1), Bx, M_zface);
               Real Hy_zface = MacroscopicProperties::getH_Maxwell(i, j, k, 1, amrex::IntVect(0,1,0), amrex::IntVect(0,0,1), By, M_zface);
               Real Hz_zface = MacroscopicProperties::getH_Maxwell(i, j, k, 2, amrex::IntVect(0,0,1), amrex::IntVect(0,0,1), Bz, M_zface);
-              // H_bias
+
+	      // H_bias
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp]:
```suggestion
	        // H_bias
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp [hunk 19]
```diff
@@ -212,21 +223,22 @@ void FiniteDifferenceSolver::MacroscopicEvolveM (
               // when working on M_zface(i,j,k,0:2) we have direct access to M_zface(i,j,k,0:2) and Hz(i,j,k)
               // Hy and Hz can be acquired by interpolation
               // H_maxwell
-              Real Hx_zface = MacroscopicProperties::getH_Maxwell(i, j, k, 0, amrex::IntVect(1,0,0), amrex::IntVect(0,0,1), Bx, M_zface);
+	      Real Hx_zface = MacroscopicProperties::getH_Maxwell(i, j, k, 0, amrex::IntVect(1,0,0), amrex::IntVect(0,0,1), Bx, M_zface);
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp]:
```suggestion
	        Real Hx_zface = MacroscopicProperties::getH_Maxwell(i, j, k, 0, amrex::IntVect(1,0,0), amrex::IntVect(0,0,1), Bx, M_zface);
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp [hunk 20]
```diff
@@ -160,12 +171,12 @@ void FiniteDifferenceSolver::MacroscopicEvolveM (
               Real Hz_bias_yface = MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(0,0,1), amrex::IntVect(0,1,0), Hz_bias);
               // H_eff = H_maxwell + H_bias + H_exchange + H_anisotropy ... (only the first two terms are considered here)
               // keep in mind to ADD H_Maxwell back to H_eff as shown below
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp]:
```suggestion
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp [hunk 21]
```diff
@@ -160,12 +171,12 @@ void FiniteDifferenceSolver::MacroscopicEvolveM (
...
-              // Real Hz_eff = Hz_yface + Hz_bias_yface;
-              Real Hx_eff = Hx_bias_yface;
-              Real Hy_eff = Hy_bias_yface;
-              Real Hz_eff = Hz_bias_yface;
+              Real Hx_eff = Hx_yface + Hx_bias_yface;
+              Real Hy_eff = Hy_yface + Hy_bias_yface;
+              Real Hz_eff = Hz_yface + Hz_bias_yface;
+              // Real Hx_eff = Hx_bias_yface;
+              // Real Hy_eff = Hy_bias_yface;
+              // Real Hz_eff = Hz_bias_yface;
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp]:
```suggestion
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp [hunk 22]
```diff
@@ -490,23 +472,18 @@ void FiniteDifferenceSolver::MacroscopicEvolveM_2nd (
...
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp]:
```suggestion
              // H_maxwell
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp [hunk 23]
```diff
@@ -426,23 +408,23 @@ void FiniteDifferenceSolver::MacroscopicEvolveM_2nd (
...
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp]:
```suggestion
              // H_maxwell
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp [hunk 24]
```diff
@@ -425,24 +407,24 @@ void FiniteDifferenceSolver::MacroscopicEvolveM_2nd (
               // Hy and Hz can be acquired by interpolation
 
               // Uncomment these lines when we couple LLG with Maxwell
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp]:
```suggestion
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp [hunk 25]
```diff
@@ -172,26 +166,25 @@ void FiniteDifferenceSolver::MacroscopicEvolveM_2nd (
...
               // H_eff = H_maxwell + H_bias + H_exchange + H_anisotropy ... (only the first two terms are considered here)
 
-              Real Hx_eff = Hx_bias_yface;
-              Real Hy_eff = Hy_bias_yface;
-              Real Hz_eff = Hz_bias_yface;
+              // Real Hx_eff = Hx_bias_yface;
+              // Real Hy_eff = Hy_bias_yface;
+              // Real Hz_eff = Hz_bias_yface;
 
               // Uncomment these lines when we couple LLG with Maxwell
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp:184]:
```suggestion
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp [hunk 26]
```diff
@@ -360,24 +347,19 @@ void FiniteDifferenceSolver::MacroscopicEvolveM_2nd (
               // Hy and Hz can be acquired by interpolation
 
               // Uncomment these lines when we couple LLG with Maxwell
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp]:
```suggestion
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp [hunk 27]
```diff
@@ -425,24 +407,24 @@ void FiniteDifferenceSolver::MacroscopicEvolveM_2nd (
...
               Real Hz_bias_yface = MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(0,0,1), amrex::IntVect(0,1,0), Hz_bias);
               // H_eff = H_maxwell + H_bias + H_exchange + H_anisotropy ... (only the first two terms are considered here)
-              Real Hx_eff = Hx_bias_yface;
-              Real Hy_eff = Hy_bias_yface;
-              Real Hz_eff = Hz_bias_yface;
+              // Real Hx_eff = Hx_bias_yface;
+              // Real Hy_eff = Hy_bias_yface;
+              // Real Hz_eff = Hz_bias_yface;
 
               // Uncomment these lines when we couple LLG with Maxwell
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp:424]:
```suggestion
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp [hunk 28]
```diff
@@ -489,24 +471,19 @@ void FiniteDifferenceSolver::MacroscopicEvolveM_2nd (
               // Hy and Hz can be acquired by interpolation
 
               // Uncomment these lines when we couple LLG with Maxwell
```
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp]:
```suggestion
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp (hunk 25 above)
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp:184]:
```suggestion
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp (hunk 27 above)
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp]:
```suggestion
```

File: Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp (hunk 25 above)
Comment (ajnonaka) [inline Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp]:
```suggestion
```

**Review by ajnonaka — APPROVED** [review]
This looks good pending:
-the minor suggestions I have below (all non-functional)
-missing an inputs file to test; but we may not have that ready before your job ends Monday
Perhaps we should merge Monday anyway and let the rest of us (Reva/Andy/Jackie) proceed from there.

**Review by ajnonaka — APPROVED** [review]

**Review by RevathiJambunathan — APPROVED** [review]

**Pull request merged**